sstables_loader.cc

```diff
@@ -81,10 +84,12 @@ class send_meta_data {
         llog.trace("send_meta_data: send mf to node={}, size={}", _node, size);
         co_return co_await _sink(fmf, streaming::stream_mutation_fragments_cmd::mutation_fragment_data);
     }
-    future<> finish(bool failed) {
+    future<> finish(bool failed, bool aborted = false) {
         std::exception_ptr eptr;
         try {
-            if (failed) {
+            if (_db.features().load_and_stream_abort_rpc_message && aborted) {
+                co_await _sink(frozen_mutation_fragment(bytes_ostream()), streaming::stream_mutation_fragments_cmd::abort);
```

Comment:
Looks like we are unconditionally aborting here, if the feature is enabled. Shouldn't we check `failed` and `aborted` first?

Reply:
We are aborting iff the `aborted` is set, and it is set only in case the exception is `abort_requested_exception` otherwise, the `aborted` is `false`. On the other hand the `failed` is set in both cases, since, well, aborted state is failed. So, what is the point to check both `failed` and `aborted`?

Reply:
Ah, I missed the `&& aborted`, I thought you only check the feature flag.